Source/WebCore/Modules/model-element/HTMLModelElement.cpp

```diff
@@ -377,6 +388,37 @@ void HTMLModelElement::applyBackgroundColor(Color color)
         m_modelPlayer->setBackgroundColor(color);
 }
 
+#if ENABLE(MODEL_PROCESS)
+const DOMMatrixReadOnly& HTMLModelElement::entityTransform() const
+{
+    return m_entityTransform.get();
+}
+
+ExceptionOr<const DOMMatrixReadOnly&> HTMLModelElement::setEntityTransform(const DOMMatrixReadOnly& transform)
+{
+    auto player = m_modelPlayer;
+    if (!player) {
+        ASSERT_NOT_REACHED();
+        return DOMMatrixReadOnly::create(TransformationMatrix::identity, DOMMatrixReadOnly::Is2D::No).get();
+    }
+
+    TransformationMatrix matrix = transform.transformationMatrix();
+
+    if (!player->supportsTransform(matrix))
+        return Exception { ExceptionCode::NotSupportedError };
```

Comment:
Ideally we have tentative web-platform-tests for this.

Reply:
Agree. Some additional tests on the TransformationMatrix conversion and supportsTransform() is also desirable, and I'm going to add them next via rdar://125239025